From e86d857faf8947b798a656c281c9c2be1c42f561 Mon Sep 17 00:00:00 2001
From: Yanjiang Jin <yanjiang.jin@windriver.com>
Date: Thu, 3 May 2012 17:29:22 +0800
Subject: [PATCH] nlm_xlp_64_be: add two syscalls

Add two XLP Specific fast system calls:
nlm_fs_mem_read16 and nlm_fs_mem_write16

Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
---
 arch/mips/include/asm/netlogic/nlm_fs.h |    4 ++++
 arch/mips/kernel/nlm_fs.c               |    2 ++
 2 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/arch/mips/include/asm/netlogic/nlm_fs.h b/arch/mips/include/asm/netlogic/nlm_fs.h
index c973e24..b49541e 100644
--- a/arch/mips/include/asm/netlogic/nlm_fs.h
+++ b/arch/mips/include/asm/netlogic/nlm_fs.h
@@ -26,9 +26,13 @@
 #define NLM_FS_MEM_WRITE32        22
 #define NLM_FS_MSGSND3            23
 #define NLM_FS_MSGRCV1            24
+#define NLM_FS_MEM_READ16         25
+#define NLM_FS_MEM_WRITE16        26
 
 #ifndef __ASSEMBLY__
 
+extern void nlm_fs_mem_read16(void);
+extern void nlm_fs_mem_write16(void);
 extern void nlm_fs_mem_read32(void);
 extern void nlm_fs_mem_write32(void);
 extern void nlm_fs_mem_read64(void);
diff --git a/arch/mips/kernel/nlm_fs.c b/arch/mips/kernel/nlm_fs.c
index 627dde4..3cb8b9c 100644
--- a/arch/mips/kernel/nlm_fs.c
+++ b/arch/mips/kernel/nlm_fs.c
@@ -35,5 +35,7 @@ unsigned long nlm_fs_table[] = {
 	[NLM_FS_PROMINFO]         = (unsigned long)nlm_fs_unused,
 	[NLM_FS_MSGSND3]          = (unsigned long)nlm_fs_msgsnd3,
 	[NLM_FS_MSGRCV1]          = (unsigned long)nlm_fs_msgrcv1,
+	[NLM_FS_MEM_READ16]       = (unsigned long)nlm_fs_mem_read16,
+	[NLM_FS_MEM_WRITE16]      = (unsigned long)nlm_fs_mem_write16,
 	0
 };
-- 
1.7.0.4

